core/api/package.json

```diff
@@ -182,6 +183,7 @@
     "pino-pretty": "^10.3.1",
     "prettier": "^3.2.5",
     "protoc-gen-js": "^3.21.2",
+    "puppeteer": "^22.7.1",
```

Comment:
not sure about this.. integrations tests are already flacky and now introducing a package like this that runs a headless chromium can be problematic.. is not another way to test it?

Reply:
I understand your point. 

I tried with axios and basically spent a day on this, and I could not make it work, so I gave up.

the issue with a OAuth2 flow is that there is a lot of context carried around different screen, whether it's via parameters over the URL or via Cookie. it's quite hard to replicate this, specially for someone not having the full context of OAuth2 in its head. 

using puppeteer make it a lot more straightforward.

I suggest we could try it and revert/change if this introduce flackiness to the test suite?